source/heating_model/interface.cc

```diff
@@ -228,6 +228,13 @@ namespace aspect
                             const MaterialModel::MaterialModelOutputs<dim> &material_model_outputs,
                             HeatingModel::HeatingModelOutputs &heating_model_outputs) const
     {
+      // Initialize the outputs to zero
```

Comment:
Maybe you can improve this comment to explain why we have to do this here (as the constructor already sets them to zero). Something like "Set the outputs to zero before evaluating the individual plugins, because the HeatingModelOutputs object is re-used for every cell and the individual plugins only add terms to the initial value." 
And then you can also set them to NaNs in the constructor, which makes it more similar to the MaterialModelOutputs.